x-pack/legacy/plugins/siem/public/pages/detection_engine/rules/components/rule_switch/index.tsx

```diff
@@ -62,13 +64,29 @@ export const RuleSwitchComponent = ({
         await enableRulesAction([id], event.target.checked!, dispatch, dispatchToaster);
```

Comment:
This is the normal path that catches/toasters the error properly. the `else` clause below is handling a case where we don't have dispatch. @XavierM is the dispatchless branch for portability, or do we have a specific use case for it right now?

Reply:
After discussing with @XavierM, it looks like the `else` path here is used on the rule details page. We should consolidate these, but they're both currently used.